goinp/goinp.go

```diff
@@ -64,45 +65,72 @@ func AskForString(messageToPrint string) (string, error) {
 	return AskForStringFromReader(messageToPrint, os.Stdin)
 }
 
-func writeStdinClearable(text string) error {
+type stdin interface {
+	Fd() uintptr
+	Read([]byte) (int, error)
+}
+
+func writeStdinClearable(text string, in stdin) error {
 	// clearable stdin text is only possible if there is an available terminal input buffer
-	if terminal.IsTerminal(int(os.Stdin.Fd())) {
+	if terminal.IsTerminal(int(in.Fd())) {
 		for _, c := range []byte(text) {
-			if _, _, errno := syscall.Syscall(syscall.SYS_IOCTL, os.Stdin.Fd(), syscall.TIOCSTI, uintptr(unsafe.Pointer(&c))); errno != 0 {
+			if _, _, errno := syscall.Syscall(syscall.SYS_IOCTL, in.Fd(), syscall.TIOCSTI, uintptr(unsafe.Pointer(&c))); errno != 0 {
 				return fmt.Errorf("failed to write to stdin, err no: %d", errno)
 			}
 		}
 	}
 	return nil
 }
 
-func askOptionsWithReader(title string, defaultValue string, optional bool, reader *bufio.Reader, options ...string) (string, error) {
-	const customValueOptionText = "<custom value>"
+func readUntil(reader stdin, str string) (string, error) {
+	var strb []byte
+	for {
+		b := make([]byte, 1)
+		if _, err := reader.Read(b); err != nil {
+			return "", err
+		}
+		strb = append(strb, b[0])
+		if bytes.HasSuffix(strb, []byte(str)) {
+			return string(strb), nil
+		}
+	}
+}
 
-	if len(options) == 0 {
-		for {
-			if title != "" {
-				fmt.Print("Enter value for \"" + strings.TrimSuffix(title, ":") + "\": ")
+func askForInput(title string, defaultValue string, optional bool, reader stdin, writer io.Writer) (string, error) {
+	for {
+		if title != "" {
+			if _, err := fmt.Fprint(writer, "Enter value for \""+strings.TrimSuffix(title, ":")+"\": "); err != nil {
```

Comment:
why do you modify the provided title? i mean other functions in this package doing this: https://github.com/bitrise-io/goinp/pull/8/files/5ce02bb46b6b52530af2a85c5f7e28733ad4a932..118e899d591475a3f899e52a2c4a5edb48c0a7d3#diff-b32312e9d7785bea2b0a3c9e4708c4ebL27

Reply:
this is for the case if someone inputs a title with a `:` at the end, do you think it is not worth to remove it? it is just wanted to be to make sure that there is no `::` at the end. btw 👍

Reply:
to be honest, i not happy with that either since i have seen multiple time questions like: `question: :`
i would let the client decide how to phrase the question since this is a core package i would keep it as simple and as flexible as possible.

other than that you, even more, restrict the question with the `Enter value for` prefix.